pkg/messaging/tracing/publisher.go

```diff
@@ -0,0 +1,42 @@
+package tracing
+
+import (
+	"context"
+
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+// traced ops
+const publishOP = "publish_op"
+
+var _ messaging.Publisher = (*publisherMiddleware)(nil)
+
+type publisherMiddleware struct {
+	publisher messaging.Publisher
+	tracer    opentracing.Tracer
+}
+
+func New(publisher messaging.Publisher, tracer opentracing.Tracer) *publisherMiddleware {
+	return &publisherMiddleware{
+		publisher: publisher,
+		tracer:    tracer,
+	}
+}
+
+// Publish implements messaging.Publisher
+func (pm *publisherMiddleware) Publish(ctx context.Context, topic string, msg *messaging.Message) error {
+	span, _ := opentracing.StartSpanFromContextWithTracer(ctx, pm.tracer, publishOP)
+	ext.MessageBusDestination.Set(span, msg.Subtopic)
```

Comment:
Add more info such as:
```go
span.SetTag("publisher", msg.Publisher)
span.SetTag("protocol", msg.Protocol)
span.SetTag("topic", topic)
```

Reply:
For publisher I don't think we need `prototcol`. There is a bit of redundacy. The traces are from the coap adapter so the protocol will be `coap`

![image](https://user-images.githubusercontent.com/28790446/232002975-08331f5c-d800-463b-b7db-687b75120ab5.png)